airflow-core/src/airflow/serialization/typing.py

```diff
@@ -29,4 +29,11 @@ def is_pydantic_model(cls: Any) -> bool:
     """
     # __pydantic_fields__ is always present on Pydantic V2 models and is a dict[str, FieldInfo]
     # __pydantic_validator__ is an internal validator object, always set after model build
-    return hasattr(cls, "__pydantic_fields__") and hasattr(cls, "__pydantic_validator__")
+    # Check if it is not a dataclass to prevent detecting pydantic dataclasses as pydantic models
+    from dataclasses import is_dataclass
```

Comment:
Please move this to top-level import.

Reply:
I agree with @uranusjr , let's keep it at the top.

Reply:
Cool, that's ok, no objections